src/CommunityToolkit.Maui.Core/Primitives/Folder.shared.cs

```diff
@@ -0,0 +1,17 @@
+namespace CommunityToolkit.Maui.Core.Primitives;
+
+/// <summary>
+/// Represents a folder in the file system.
+/// </summary>
+public class Folder
+{
+	/// <summary>
+	/// Folder path.
+	/// </summary>
+	public string Path { get; init; } = null!;
+
+	/// <summary>
+	/// Folder name.
+	/// </summary>
+	public string Name { get; init; } = null!;
```

Comment:
Let's avoid the null-forging operator, `!`.

```suggestion
	public string Name { get; init; } = string.Empty;
```